stratosphere/pm/source/boot2/boot2_api.cpp

```diff
@@ -0,0 +1,294 @@
+/*
+ * Copyright (c) 2018-2019 Atmosphère-NX
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms and conditions of the GNU General Public License,
+ * version 2, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <cctype>
+#include <dirent.h>
+#include <stratosphere/cfg.hpp>
+#include <stratosphere/ldr.hpp>
+#include <stratosphere/pm.hpp>
+#include "boot2_api.hpp"
+
+namespace sts::boot2 {
+
+    namespace {
+
+        /* Launch lists. */
+
+        /* psc, bus, pcv is the minimal set of required titles to get SD card. */
+        /* bus depends on pcie, and pcv depends on settings. */
+        constexpr ncm::TitleId PreSdCardLaunchPrograms[] = {
+            ncm::TitleId{TitleId_Psc},         /* psc */
+            ncm::TitleId{TitleId_Pcie},        /* pcie */
+            ncm::TitleId{TitleId_Bus},         /* bus */
+            ncm::TitleId{TitleId_Settings},    /* settings */
+            ncm::TitleId{TitleId_Pcv},         /* pcv */
+        };
+        constexpr size_t NumPreSdCardLaunchPrograms = sizeof(PreSdCardLaunchPrograms) / sizeof(PreSdCardLaunchPrograms[0]);
```

Comment:
I would suggest adding a template-based countof helper to libstrat.

Reply:
Yeah, it'd do well in defines.hpp.